app/lib/actions/pulp/abstract_async_task.rb

```diff
@@ -79,7 +79,7 @@ def external_task
         output[:pulp_tasks]
       end
 
-      def cancel!
+      def cancel!(raise_error = true)
```

Comment:
Yes, so maybe a better term would be better  'raise_error_on_cancel' or 'fail_on_cancel'

Raising an error is somewhat the intent of the cancel! function as we've implemented it (its 'inherited' from the cancellable module in dynflow).  The other option would be to move most of the logic to some other function  'cancel'  and then just call that from the timeout methods and 'cancel!'.  In some ways i could see that being more confusing, in some ways less (as this is more explicit about what is going on)  so I will let you give your input :)